default.nix

```diff
@@ -103,6 +104,203 @@ in obApp // rec {
+      gi-javascriptcore = addGObjectIntrospection (self.callHackage "gi-javascriptcore" "4.0.20" { webkitgtk = ubuntuWebkitgtk; });
+      gi-webkit2 = addGObjectIntrospection (self.callHackage "gi-webkit2" "4.0.24" { webkitgtk = ubuntuWebkitgtk; });
+      webkitgtk3-javascriptcore = addGObjectIntrospection (self.callHackage "webkitgtk3-javascriptcore" "0.14.2.1" { webkitgtk = ubuntuWebkitgtk; });
+      linux = pkgs.haskell.lib.overrideCabal super.linux (old: {
+        libraryPkgconfigDepends = [ ubuntuWebkitgtk pkgs.glib-networking ];
+      });
+    };
+  };
+  ubuntuExe = pkgs.haskell.lib.justStaticExecutables ubuntuHPkgs.linux;
+  # Webkitgtk hard compiles the libexec path into the library, so we have to patch it if we
+  # want it to work not from the nix store.
+  ubuntuWebkitgtk = pkgs.webkitgtk.overrideAttrs (old: {
+    patches = old.patches ++ [
+      # Note this patch overrides the macro, because we can't do it in the cmake file lol
+      # This wrecks anything we try to do to reference outside the nix store. 
+      # https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L10
+      # Also in newer webgtks there is some sandboxing going on, which means that we may have to fiddle with another patch
+      # when that happens. See https://github.com/NixOS/nixpkgs/commit/84fb39ef12a71caabd4fb4a87a4892497bcb2f7f
+      ./patches/ubuntu/webkitgtk/ubuntupaths.patch
+    ];
+  });
+  deb = pkgs.stdenv.mkDerivation {
+    name = "${linuxAppName}-usrlib";
+    outputs = ["out"];
+    nativeBuildInputs = [ pkgs.makeWrapper ];
+    exportReferencesGraph = ["graph" ubuntuExe];
+    builder = pkgs.writeScript "builder.sh" ''
+      source $stdenv/setup
+      set -ex
+      mkdir $out
+      export LIBPATH=/usr/lib/chainweaver
+      export LIBEXECPATH=/usr/libexec/chainweaver
+      export BINPATH=/usr/bin
+      export DEBDIR=$TMPDIR/${linuxAppName}-1
+      export BINDIR=$DEBDIR$BINPATH
+      export LIBEXECDIR=$DEBDIR$LIBEXECPATH
+      export LIBDIR=$DEBDIR$LIBPATH
+      export TMPEXE=$TMPDIR/${linuxAppName}
+      export CHAINWEAVER_LIBEXEC_EXE=$LIBEXECDIR/${linuxAppName}
+      export CHAINWEAVER_BIN_EXE=$BINDIR/${linuxAppName}
+
+      # We want to patch a bunch of stuff to use the ubuntu things
+      function chainweaver_patchelf () {
+        patchelf --set-interpreter /lib64/ld-linux-x86-64.so.2 $1
+        patchelf --set-rpath $LIBPATH $1
+      }
+
+      # make debian control file structure
+      mkdir -p $DEBDIR/DEBIAN
+      cp ${deb-control} $DEBDIR/DEBIAN/control
+      cp ${deb-copyright} $DEBDIR/DEBIAN/copyright
+      cp ${deb-changelog} $DEBDIR/DEBIAN/changelog
+
+      mkdir -p $BINDIR
+      mkdir -p $LIBDIR
+      mkdir -p $LIBEXECDIR
+
+      cp ${obApp.ghc.linux}/bin/linuxApp $TMPEXE
+      chmod u+w $TMPEXE
+       
+      cp $TMPEXE $CHAINWEAVER_LIBEXEC_EXE
+      cp "${pkgs.z3}"/bin/z3 "$LIBEXECDIR/z3"
+
+      #TODO : This shouldn't be in libexec
```

Comment:
Yeah, obelisk by default looks in executable directory for resources. You can use something like this to make it configurable via a cli flag:

- https://github.com/obsidiansystems/obelisk/pull/558
- https://gitlab.com/obsidian.systems/kiln/merge_requests/733/diffs#diff-content-5b1d1ec64e71b88e93f64a6e7dbb173c49239c6e